scala/scala.bzl

```diff
@@ -72,11 +73,30 @@ def _compile(ctx, jars, buildijar):
       ijar=ctx.file._ijar.path,
       out=ctx.outputs.jar.path,
       ijar_out=ctx.outputs.ijar.path)
+  #TODO here is the issue...when scala library depends on this, it's not picking up anything
+
+  #TODO I need a command that will explode out the source jars
+  #TODO probably can be in a helper function
+  srcjar_cmd = "\n_SRCJAR_ARG=\" \""
+  srcjars = _srcjar_filetype.filter(ctx.files.srcjars + list(_collect_srcjars(ctx.attr.deps)))
+  if len(srcjars) > 0:
+    # TODO Somehow, the manifest is sneaking in here?! so we filter... but really
+    #      should try and understand why the manifest is being put here!
+    # TODO the FileType.filter also seemed to get rid of it...??
+    for srcjar in srcjars:
+      srcjar_cmd += """
+mkdir -p {{out}}_tmp/expand_srcjars/{srcjar}
+unzip {srcjar} -d {{out}}_tmp/expand_srcjars/{srcjar}
+""".format(srcjar = srcjar.path)
+    srcjar_cmd += """
+for f in `find {out}_tmp/expand_srcjars -type f -name "*.scala"`; do _SRCJAR_ARG=$_SRCJAR_ARG$f" "; done
```

Comment:
yes...wasn't sure how else to do it. In a sense, bash is kind of like a monad you can't get out of... we can put information into it, but we can't get it out. I'm not sure outside of bash how we can say "we know that this is an archive with files in it, what are all the files?" Of course, an option would be to wrap scrooge as you suggested (this is not a bad option!), but for now this is all I could come up with.